sbin/prepareWorkspace.sh

```diff
@@ -79,6 +122,8 @@ checkoutAndCloneOpenJDKGitRepo() {
       echo "If this is inside a docker you can purge the existing source by passing --clean-docker-build"
       exit 1
     fi
+  elif [ "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE]}" == "true" ]; then
+    copyFromDirOrUnpackFromArchive
```

Comment:
It feels a little odd to be calling this from within `checkoutAndCloneOpenJDKGitRepo` - perhaps we could invoke it from the higher level configureWorkspace` instead as an alternative. I guess it depends how much of the extra logic in this function we'd need to duplicate or abstract out.

Reply:
ok. let me try.

Reply:
Oh I see.  There is
```
  cd "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}"

  # Check that we have a git repo, we assume that it is a repo that contains openjdk source
  if [ -d "${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}/.git" ] && [ "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE]}" == "false" ]; then
```

So the change of control dir, which is pretty important for any future steps would need to be handled.
And the main if is still using the `[ "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE]}" == "false" ]` 

becasue of the change of the dir, I would rather not chnage the location of the call, unless yo really insists.

In that case it will go :
```
function configureWorkspace() {
  if [[ "${BUILD_CONFIG[ASSEMBLE_EXPLODED_IMAGE]}" != "true" ]]; then
    createWorkspace
    downloadingRequiredDependencies
    downloadDevkit
    relocateToTmpIfNeeded
+if [ "${BUILD_CONFIG[OPENJDK_LOCAL_SOURCE_ARCHIVE]}" == "true" ]; then
+    copyFromDirOrUnpackFromArchiveWrapper
+else 
    checkoutAndCloneOpenJDKGitRepo
+fi
    applyPatches
    if [ "${BUILD_CONFIG[CUSTOM_CACERTS]}" = "true" ] ; then
      prepareMozillaCacerts
    fi
  fi
  writeDockerImageSHA
}
```
where the  wrapper:
```
function   copyFromDirOrUnpackFromArchiveWrapper() {
   cd "${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}"
   copyFromDirOrUnpackFromArchive
    cd "${BUILD_CONFIG[WORKSPACE_DIR]}"
}
```

IN addition the method of `checkoutAndCloneOpenJDKGitRepo` is doing a bit more before `  cd "${BUILD_CONFIG[WORKSPACE_DIR]}"`   Although  it seems those lines looks oook to omit (temurin exit, update openj9 and createSourceTagFile ) I'm not sure If I feel safe to omit the,

Reply:
Yeah agree that the safest option is to leave it as-is although putting it in the `configureWorkspace` function as per your example does look a lot nicer. But I won't insist on it :-)